seatunnel-connectors-v2/connector-phoenix/src/main/java/org/apache/seatunnel/connectors/seatunnel/phoenix/client/PhoenixStatementExecutor.java

```diff
@@ -0,0 +1,281 @@
+        }
+        List<Integer> sinkColumnsIndex = phoenixWriteConfig.getSinkColumnsIndexInRow();
+        if (columnTypes == null) {
+            // no types provided
+            for (int index = 0; index < sinkColumnsIndex.size(); index++) {
+                upload.setObject(index + 1, row.getFields()[index]);
+            }
+        } else {
+            // types provided
+            for (int i = 0; i < sinkColumnsIndex.size(); i++) {
+                int sqlType = columnTypes[i];
+                setupColumn(upload, i + 1, sqlType, row.getFields()[i]);
+            }
+        }
+    }
+
+    private void setupColumn(PreparedStatement ps, int pos, int sqlType, Object col) throws SQLException {
+        if (col != null) {
+            switch (sqlType) {
+                case Types.CHAR:
+                case Types.VARCHAR:
+                    ps.setString(pos, (String) col);
+                    break;
+
+                case Types.BINARY:
+                case Types.VARBINARY:
+                    ps.setBytes(pos, (byte[]) col);
+                    break;
+
+                case Types.BOOLEAN:
+                    ps.setBoolean(pos, (Boolean) col);
+                    break;
+
+                case Types.TINYINT:
+                case Constant.TYPE_UNSIGNED_TINYINT:
+                    ps.setByte(pos, ((Long) col).byteValue());
+                    break;
+
+                case Types.SMALLINT:
+                case Constant.TYPE_UNSIGNED_SMALLINT:
+                    ps.setShort(pos, ((Long) col).shortValue());
+                    break;
+
+                case Types.INTEGER:
+                case Constant.TYPE_UNSIGNED_INTEGER:
+                    ps.setInt(pos, ((Long) col).intValue());
+                    break;
+
+                case Types.BIGINT:
+                case Constant.TYPE_UNSIGNED_LONG:
+                    ps.setLong(pos, (Long) col);
+                    break;
+
+                case Types.FLOAT:
+                    ps.setFloat(pos, ((Double) col).floatValue());
+                    break;
+
+                case Types.DOUBLE:
+                    ps.setDouble(pos, (Double) col);
+                    break;
+
+                case Types.DECIMAL:
+                    ps.setBigDecimal(pos, (BigDecimal) col);
+                    break;
+
+                case Types.DATE:
+                case Constant.TYPE_UNSIGNED_DATE:
+                    ps.setDate(pos, new java.sql.Date(new Date((Long) col).getTime()));
+                    break;
+
+                case Types.TIME:
+                case Constant.TYPE_UNSIGNED_TIME:
+                    ps.setTime(pos, new java.sql.Time(new Date((Long) col).getTime()));
+                    break;
+
+                case Types.TIMESTAMP:
+                case Constant.TYPE_UNSIGNED_TIMESTAMP:
+                    ps.setTimestamp(pos, new java.sql.Timestamp(new Date((Long) col).getTime()));
+                    break;
```

Comment:
The time object is processed as a Local* time class in Seatunnel.
you can see https://github.com/apache/incubator-seatunnel/issues/2024 or source code of data type.

Reply:
> The time object is processed as a Local* time class in Seatunnel. you can see #2024 or source code of data type.

Thank you for your suggestion, I has been fixed in the new commit, refer to https://phoenix.apache.org/language/datatypes.html , transform seatunnel datatype  supported by Phoenix